pandas/io/common.py

```diff
@@ -170,7 +170,7 @@ def _expand_user(filepath_or_buffer: str | BaseBufferT) -> str | BaseBufferT:
     return filepath_or_buffer
 
 
-def validate_header_arg(header) -> None:
+def validate_header_arg(header: object) -> None:
```

Comment:
should be `Any`?

Reply:
Typeshed recommends to use `Any` when it is too difficult to create a type annotations. `object` should be used if literally any object can be accepted.

https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#conventions

> Note that Any is not the correct type to use if you want to indicate that some function can accept literally anything: in those cases use object instead.